File: 30 Days of Code/Day 12 Inheritance/Student.java
class Student extends Person{
	private int[] testScores;

    public Student(String first, String last, int id, int[] scores) {
        super(first, last, id);
        this.testScores = scores;
    }

    public char calculate() {
        int total = 0;
        for (int i = 0; i < testScores.length; i++) {
            total += testScores[i];
        }

        int average = total / testScores.length;
        char score = 'T';

        if (average > 100) {
            System.out.println("Score can't over 100!");
        } else if (90 <= average) {
            score = 'O';
        } else if (80 <= average) {
            score = 'E';
        } else if (70 <= average) {
            score = 'A';
        } else if (55 <= average) {
            score = 'P';
        } else if (40 <= average) {
            score = 'D';
        }
        return score;
    }
}

File: 30 Days of Code/Day 12 Inheritance/student.py
class Student(Person):
    def __init__(self, first, last, idNum, scores):
        super().__init__(first, last, idNum)
        self.scores = scores[:]

    def calculate(self):
        total = sum(self.scores)
        average = total/len(self.scores)
        if average > 100:
            print("Score cannot over 100!")
        elif 90 <= average:
            return "O"
        elif 80 <= average:
            return "E"
        elif 70 <= average:
            return "A"
        elif 55 <= average:
            return "P"
        elif 40 <= average:
            return "D"
        else:
            return "T"
